### PR TITLE
Improve Build Flexibility with `pkg-config` for Better Dynamic Linking

### PR DESCRIPTION
**Pull request by fkasischke**
#### Objective
The primary objective of this PR is to address the challenge of manually configuring environment variables when using system libraries in the `ort-sys` crate. This process can be error-prone and cumbersome, especially when ensuring compatibility across different development environments. By integrating `pkg-config`, it aims to automate and streamline the detection and configuration of the ONNX Runtime library and other dependencies, enhancing developer productivity and improving build consistency.

#### Motivation
Currently, developers/users need to manually set environment variables to configure the correct paths for linking system libraries. This manual setup not only slows down the development process but also introduces potential errors and inconsistencies across different platforms. Automating this setup through `pkg-config` will minimize human error, speed up the build process, and ensure that the correct library versions and configurations are used.

#### What Changes?
1. **Integration of `pkg-config`**:
   - Added `pkg-config` as a dependency in `Cargo.toml`.
   - Implemented a new function `try_setup_with_pkg_config()` in `build.rs`, which leverages `pkg-config` to automatically find and configure the necessary compiler and linker settings for the ONNX Runtime system lib.
   - If `pkg-config` finds the lib, it automatically sets up the build environment, in case it is not successful it uses the existing manual configuration.

3. **Dynamic Linking Enhancement**:
   - **Dynamic Libraries**: By leveraging `pkg-config`, the build script automates the process of configuring dynamic linker flags. 

#### Testing
- Tested it across multiple platforms (Linux and macOS) that it works with `pkg-config` and with the existing manual fallback

### Todo
- [x] Test on Windows
- [x] Update docs

### CODE REVIEW

File: ort-sys/build.rs [hunk 1]
```diff
@@ -376,6 +376,19 @@ fn prepare_libort_dir() -> (PathBuf, bool) {
 	}
 }
 
+fn try_setup_with_pkg_config() -> bool {
+	match pkg_config::Config::new().probe("libonnxruntime") {
+		Ok(lib) => {
```
Comment (decahedron1) [inline ort-sys/build.rs:381]: Shouldn't this be using at least `lib.link_paths` here? FWICT, `probe` seems to only return the package configuration, it doesn't configure the link args

**Comment by decahedron1** [comment]
Windows looks good, so if you could update `docs/setup/linking.mdx`, LGTM.

File: ort-sys/build.rs [hunk 2]
```diff
@@ -376,6 +376,19 @@ fn prepare_libort_dir() -> (PathBuf, bool) {
 	}
 }
 
+fn try_setup_with_pkg_config() -> bool {
+	match pkg_config::Config::new().probe("libonnxruntime") {
```
Comment (decahedron1) [inline ort-sys/build.rs:380]: Also think this should be `onnxruntime` and not `libonnxruntime`

File: ort-sys/build.rs (hunk 1 above)
Comment (fkasischke) [reply ort-sys/build.rs:381]: Thanks for the hint, you are right. Fixed it.

**Comment by fkasischke** [comment]
> Windows looks good, so if you could update `docs/setup/linking.mdx`, LGTM.
Updated the docs

Thanks for the review @decahedron1! Could you take another look?

**Comment by fkasischke** [comment]
@decahedron1 would you mind to take another look?

**Comment by fkasischke** [comment]
Could someone of the main contributors take another look and ✅, pls?

**Comment by decahedron1** [comment]
Thank you for your contribution, unfortunately I don't think I can merge this PR since Mintlify's free plan only allows one editor, and since you modified the docs, it's counting you as a 2nd editor and refusing to update the documentation website unless I upgrade my plan 🙃

Please reopen with just the changes in `ort-sys/`, I'll update the docs separately.

**Pull request closed without merging**